### PR TITLE
Kennric/database config

### PR DESCRIPTION
**Pull request by Kennric**
fixes issue #9

## Changes in this PR.
- [X] Creates test/dev/prod environment based settings
- [X] Connects the models to the env appropriate database
- [X] Sets up in-memory database for testing

## Testing this PR.
1. Run rake spec
2. See model tests pass

@osuosl/devs

### CODE REVIEW

**Comment by ElijahCaine** [comment]
```
[root@29ec38b3fbfd code]# rake spec
/usr/local/bin/ruby -I/usr/local/lib/ruby/gems/2.3.0/gems/rspec-core-3.4.4/lib:/usr/local/lib/ruby/gems/2.3.0/gems/rspec-support-3.4.1/lib /usr/local/lib/ruby/gems/2.3.0/gems/rspec-core-3.4.4/exe/rspec spec/models/client_spec.rb --format documentation

The Client Model and table
  can create a client
  has no clients
  has a name

Finished in 0.00907 seconds (files took 0.97033 seconds to load)
3 examples, 0 failures
```
lgtm +1

File: Gemfile [hunk 1]
```diff
@@ -1,14 +1,39 @@
...
+  # Tools
+  gem 'dotenv'
+  gem 'thin'
+  gem 'json'
+  gem 'redis', '~>3.2'
+  gem 'rubocop', require: false
+end
+
+# Development
+group :production do
```
Comment (alxngyn) [inline Gemfile]: I assume it should be `:development`

**Comment by alxngyn** [comment]
Aside from the little thing in Gemfile, looks great!
+1

**Comment by Kennric** [comment]
Fixed mis-reference to production

**Pull request merged**